### ISSUE TITLE
Custom metrics doc doesn't mention `assign` method

### ISSUE THREAD

**Issue by nicolasnn** [tensorflow/tensorflow#51862]
## URL(s) with the issue:
https://www.tensorflow.org/guide/keras/train_and_evaluate#custom_metrics

## Description of issue (what needs changing):

**First small issue**:  `reset_states` is deprecated, il should be `reset_state`, shouldn't it? Furthermore, this method is not required when creating a custom metrics.

**Second issue**:
It is not necessary to subclass Metric (cf https://github.com/tensorflow/tensorflow/issues/28601#issuecomment-505098700). This should be written in the documentation. I am sure many users would be interested by the "easy" way, i.e. passing a function `def my_metric(y_true, y_pred)` as a metric

**Third issue**:
The doc shows an example with `assign_add` when updating the metrics. However, it may not fit users need.
For example, let's consider I want to compute a Peak Signal to Noise Ratio metric (a kind of logarithmic MSE). If I follow the doc, I would write something like this: 

```python
import tensorflow as tf
from tensorflow import keras
import math

class PSNR(keras.metrics.Metric):
    """
    Peak Signal to Noise Ratio metric
    """
    def __init__(self, name='PSNR', **kwargs):
        super().__init__(name, **kwargs)
        self.psnr = self.add_weight(name='PSNR', initializer='zeros')

    def update_state(self, y_true, y_pred, sample_weight=None):
        y_pred, y_true = tf.cast(y_pred, tf.float32), tf.cast(y_true, tf.float32)
        mse = tf.reduce_mean(keras.metrics.mean_squared_error(y_true, y_pred))
        psnr = 10.0 * tf.divide(tf.math.log(tf.divide(10000**2, mse)), math.log(10))

        self.psnr.assign_add(psnr)

    def result(self):
        return self.psnr
```

As a high level API user, I am not familiar with `tf.Variable` methods. Thus, I used `assign_add` like in the documentation instead of `assign` because I had no clue `assign` existed. And it took me some time to figure why my metric was so high and increasing so fast...

### Clear description
- The documentation should mention the "easy" way to create a metric, i.e. passing a function `def my_metric(y_true, y_pred)`. It should be quite the same as the custom losses section (https://www.tensorflow.org/guide/keras/train_and_evaluate#custom_losses)
- I think an exemple with `assign` would benefit some users who would need to implement a metric that is not a sum

### Correct links
Yes

### Parameters defined
Yes

### Returns defined
Yes

### Raises listed and defined
Yes

### Usage example
Yes

### Request visuals, if applicable
No

### Submit a pull request?
don't know

**Comment by tilakrayal** [comment]
@nicolasnn ,
We see that the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose) has not been filled, could you please do so as it helps us analyse the issue.Thanks!

**Comment by nicolasnn** [comment]
Edited

**Comment by MrinalTyagi** [comment]
@tilakrayal @sanatmpa1 i would like to work on this issue. Could you guide me regarding the same?

**Comment by sachinprasadhs** [comment]
Will take your issue one by one as follow.

1. You are right `reset_states` has been depreciated, instead you can use `reset_state` as mentioned in the warning, I can help you with creating PR for thew same.

2. Subclassing metric will resolve many of the complex scenarios =, whereas other methods will be having certain limitations. You can still use the simplest method to define custom metric as you have mentioned. You can also look into [this](https://medium.com/swlh/custom-loss-and-custom-metrics-using-keras-sequential-model-api-d5bcd3a4ff28) article where he has explained multiple ways to define custom metrics.

3. `assign_add` to update the metrics was used specific to the training example shown in the document, however, as per your requirement you can use any of the state ops mentioned [here](https://www.tensorflow.org/api_docs/cc/group/state-ops).
